src/main/java/Duke.java

```diff
@@ -1,10 +1,196 @@
+        Command input = new Command(in.nextLine());
+
+        //loop runs till the user inputs bye in the cli
+        while (true) {
+            try {
+                CommandExecute type = input.extractType();
+                System.out.println(LINE_SEPARATOR);
+                switch (type) {
+                case MARK_DONE:
+                    int taskNumber = extractTaskNumber(input.getMessage());
+                    taskDone(taskList, taskNumber);
+                    break;
+                case LIST:
+                    //print list of items
+                    TaskManager.listTasks();
+                    break;
+                case TODO:
+                    String toDoTask = input.getMessage();
+                    addToDo(taskList, toDoTask);
+                    break;
+                case DEADLINE:
+                    String deadlineTask = input.getMessage();
+                    addDeadline(taskList, deadlineTask);
+                    break;
+                case EVENT:
+                    String eventTask = input.getMessage();
+                    addEvent(taskList, eventTask);
+                    break;
+                case EXIT:
+                    Messages.exit(taskList);
+                    break;
+                case DELETE:
+                    int id = Integer.parseInt(input.getMessage().substring(7));
+                    Messages.removeTask(taskList, id);
+                    break;
+                default:
+                    Messages.printCommandNotFound();
+                    break;
+                }
+                if (type == CommandExecute.EXIT) {
+                    break;
+                }
+            } catch (DukeException e) {
+                Messages.printError(e.toString());
+            }
+            try {
+                taskList.writeToFile();
+            } catch (IOException e) {
+                Messages.printFileError();
+            }
+            System.out.println(LINE_SEPARATOR);
+            input = new Command(in.nextLine());
+        }
+    }
+
+    private static void addEvent(TaskManager taskList, String eventTask) throws DukeException {
+        String[] taskDetails = eventTask.trim().split(TASK_AT);
+        if (taskDetails[0].equals("event")) {
+            throw new DukeException(ExceptionType.MISSING_TASK_DESCRIPTION);
+        }
+        if (!eventTask.contains(TASK_AT)) {
+            throw new DukeException(ExceptionType.MISSING_IDENTIFIER);
+        }
+        if (taskDetails.length < 2) {
+            throw new DukeException(ExceptionType.MISSING_ON_TIME);
+        }
+        String description = taskDetails[0].substring(6).trim();
+        String by = taskDetails[1].trim();
+
+        Task task = TaskManager.addEvent(description, by);
+        Messages.printTaskAddedMessage(task, taskList.getTaskCount());
+    }
+
+    private static void addDeadline(TaskManager taskList, String deadlineTask) throws DukeException {
```

Comment:
Consider handling this in the deadline class

Reply:
I didn't get this, would it be better if I handle the error when a task is being added to the arraylist?